examples/chef/common/stubs.cpp

```diff
@@ -347,22 +346,101 @@ void emberAfWakeOnLanClusterInitCallback(EndpointId endpoint)
 }
 #endif
 
-void ApplicationInit()
+/**
+ * This initializer is for the application having refrigerator on EP1 and child
+ * temperatureControlledCabinet endpoints on EP2 and EP3.
+ */
+void RefrigeratorTemperatureControlledCabinetInit()
 {
-    ChipLogProgress(NotSpecified, "Chef Application Init !!!");
-
-#ifdef MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
-    // set Parent Endpoint and Composition Type for an Endpoint
     EndpointId kRefEndpointId           = 1;
     EndpointId kColdCabinetEndpointId   = 2;
     EndpointId kFreezeCabinetEndpointId = 3;
-    SetTreeCompositionForEndpoint(kRefEndpointId);
-    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
-    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
-    // set TagList
-    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
-    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
-#endif // MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
+    bool isRefrigerator;
+    bool isColdCabinet;
+    bool isFreezeCabinet;
+    if ((DataModelUtils::isEndpointOfDeviceType(kRefEndpointId, DataModelUtils::REFRIGERATOR_DEVICE_ID, isRefrigerator) ==
+         CHIP_NO_ERROR) &&
+        isRefrigerator)
+    {
+        ChipLogProgress(NotSpecified, "Refrigerator device type on EP: %d", kRefEndpointId);
+        SetTreeCompositionForEndpoint(kRefEndpointId);
+
+        // Cold Cabinet
+        if ((DataModelUtils::isEndpointOfDeviceType(kColdCabinetEndpointId,
+                                                    DataModelUtils::TEMPERATURE_CONTROLLED_CABINET_DEVICE_ID,
+                                                    isColdCabinet) == CHIP_NO_ERROR) &&
+            isColdCabinet)
+        {
+            SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
+            SetTagList(kColdCabinetEndpointId,
+                       Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(gRefrigeratorTagList));
+        }
+
+        // Freeze Cabinet
+        if ((DataModelUtils::isEndpointOfDeviceType(kFreezeCabinetEndpointId,
+                                                    DataModelUtils::TEMPERATURE_CONTROLLED_CABINET_DEVICE_ID,
+                                                    isFreezeCabinet) == CHIP_NO_ERROR) &&
+            isFreezeCabinet)
+        {
+            SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
+            SetTagList(kFreezeCabinetEndpointId,
+                       Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(gFreezerTagList));
+        }
+    }
+}
+
+/**
+ * This initializer is for the application having oven on EP1 and child endpoints -
+ * temperatureControlledCabinet on EP2, cooktop on EP3 and cooksurface on EP4.
+ */
+void OvenTemperatureControlledCabinetCooktopCookSurfaceInit()
```

Comment:
This is the oven init. There can be a split of 2: oven init and a cooktop init. The cooktop init can be given the starting endpoint ID, which can then be used for either a standalone cooktop, or a nested cooktop in an oven.